src/stacks-hierarchy/Root_PurchaseTariffZonesSearchByMapScreen.tsx

```diff
@@ -37,17 +37,17 @@ export const Root_PurchaseTariffZonesSearchByMapScreen = ({
   const {t} = useTranslation();
 
   useEffect(() => {
-    setSelectedZones({
-      ...selectedZones,
+    setSelectedZones((prev) => ({
+      ...prev,
       from: fromTariffZone,
-    });
+    }));
   }, [fromTariffZone]);
 
   useEffect(() => {
-    setSelectedZones({
-      ...selectedZones,
+    setSelectedZones((prev) => ({
+      ...prev,
       to: toTariffZone,
-    });
+    }));
   }, [toTariffZone]);
```

Comment:
Isn't it necessary to add setSelectedZones to the dependency list?

Reply:
> The set functions returned by useState also have stable identity, so you will often see them omitted from the dependencies too. If the linter lets you omit a dependency without errors, it is safe to do.

https://react.dev/learn/synchronizing-with-effects#why-was-the-ref-omitted-from-the-dependency-array